free/src/main/scala/cats/free/Free.scala

```diff
@@ -147,11 +162,9 @@ sealed abstract class Free[S[_], A] extends Product with Serializable {
    *
    * If your natural transformation is effectful, be careful. These
    * effects will be applied by `compile`.
-   */
-  final def compile[T[_]](f: FunctionK[S, T]): Free[T, A] =
-    foldMap[Free[T, ?]] { // this is safe because Free is stack safe
-      λ[FunctionK[S, Free[T, ?]]](fa => Suspend(f(fa)))
-    }(Free.catsFreeMonadForFree)
+    */
+  @deprecated("Use mapK", "1.0.0")
```

Comment:
I am not sure if we want to deprecate this one.  `compile` seems a meaningful alias.